lib/oli_web/components/delivery/actions/actions.ex

```diff
@@ -55,71 +57,98 @@ defmodule OliWeb.Components.Delivery.Actions do
       id="student_actions"
       class="mx-10 mb-10 bg-white dark:bg-gray-800 shadow-sm px-14 py-7 flex flex-col gap-6"
     >
-      <.live_component
-        module={OliWeb.Components.LiveModal}
-        id="unenroll_user_modal"
-        title="Unenroll student"
-        on_confirm={JS.push("unenroll", target: @myself)}
-      >
-        <div class="px-4">
-          <p>
-            Are you sure you want to unenroll "<%= @user.name %>" from the course "<%= @section.title %>"?
-          </p>
-        </div>
-      </.live_component>
-
-      <div class="flex flex-col sm:flex-row sm:items-end instructor_dashboard_table">
-        <h4 class="torus-h4 !py-0 mr-auto dark:text-white">Actions</h4>
-      </div>
-
-      <div class="flex justify-between items-center">
-        <div class="flex flex-col">
-          <span class="dark:text-white">Change enrolled user role</span>
-          <span class="text-xs text-gray-400 dark:text-gray-950">
-            Select the role to change for the user in this section.
-          </span>
-        </div>
-        <form phx-change="display_confirm_modal" phx-target={@myself}>
-          <select class="torus-select pr-32" name="filter_by_role_id">
-            <%= for elem <- @user_role_data do %>
-              <option selected={elem.id == @user_role_id} value={elem.id}>
-                <%= elem.title %>
-              </option>
-            <% end %>
-          </select>
-        </form>
-      </div>
-      <%= if @section.requires_payment and @is_admin do %>
-        <div class="flex justify-between items-center px-14 py-8">
-          <div class="flex flex-col">
-            <span class="dark:text-black">Bypass payment</span>
-            <span class="text-xs text-gray-400 dark:text-gray-950">Apply bypass payment</span>
-          </div>
-          <button
-            class="torus-button flex justify-center primary h-9 w-48"
-            disabled={@has_payment}
-            phx-click="display_bypass_modal"
-            phx-target={@myself}
+      <%= if @is_suspended? do %>
+        <div id="unenrolled_student_actions">
+          <.live_component
```

Comment:
NIT: instead of using a LiveComponent for the modal we can use the modal in its function component flavour `lib/oli_web/components/modal.ex`